discovery-provider/src/challenges/referral_challenge.py

```diff
@@ -0,0 +1,33 @@
+from typing import Dict, List, Optional
+
+from sqlalchemy.orm.session import Session
+from src.challenges.challenge import (
+    ChallengeManager,
+    ChallengeUpdater,
+    FullEventMetadata,
+    UserChallenge,
+)
+
+
+class ReferralChallengeUpdater(ChallengeUpdater):
```

Comment:
@piazzatron This goes back to the arguments around checking persistent storage vs trusting the event metadata. I think I would need to completely implement the `update_user_challenges` and make sure the `UserEvents` are still there, right? Filtering out those `UserChallenges` that don't have a record in storage? Should there be a hook earlier to filter the new `UserChallenge`s that are getting created, rather (or in addition to) this one that updates the `UserChallenge`s?